src/main/java/org/hrorm/SqlBuilder.java

```diff
@@ -115,10 +116,11 @@ public String selectByColumns(String ... columnNames){
     }
 
     public String selectChildIds(String parentColumn){
+
         StringBuilder buf = new StringBuilder();
 
         buf.append("select ");
-        buf.append(primaryKey.getName());
+        buf.append(primaryKey.orElseThrow(() -> new UnsupportedOperationException("DAO does not support Singular operations")).getName());
```

Comment:
This to me is a good argument for separate DaoDescriptor/KeyedDaoDescriptor classes, and perhaps a KeyedSqlBuilder variant.